WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1798,6 +1807,18 @@ public boolean updatePostContentNewEditor(boolean isAutoSave, String title, Stri
         return titleChanged || contentChanged;
     }
 
+    /*
+      * for as long as the user is in the Editor, we check whether there are any differences in media items
+      * being uploaded since they opened the Editor for this Post. If some items have finished, the current list
+      * won't be equal and thus we'll know we need to save the Post content as it's changed, given the local
+      * URLs will have been replaced with the remote ones.
+     */
+    private boolean compareCurrentMediaMarkedUploadingToOriginal(String newContent) {
+        List<String> currentUploadingMedia = AztecEditorFragment.getMediaMarkedUploadingInPostContent(this, newContent);
+        Collections.sort(currentUploadingMedia);
+        return !mMediaMarkedUploadingOnStartIds.equals(currentUploadingMedia);
```

Comment:
Since `mMediaMarkedUploadingOnStartIds` is left unchanged here, won't this method return true every time it's called if uploaded media changed at any time?

Reply:
As long as changes happen in between the user opened the Editor (which is when `mMediaMarkedUploadingOnStartIds` is populated) and exiting the Editor, that's exactly the window we want to cover, so yes it'd be fine. Unless I'm not seeing some side effect of it that could potentially be harmful?

Reply:
If a media item finished uploading between the start of the activity and the first time `updatePostContentNewEditor` is called, then every time it's called afterwards it will also return true. Looking at the flow from `savePostAsync` > `updatePostObject` > `updatePostContentNewEditor`, won't this mean that the post could be updated (and uploaded) unnecessarily?

Reply:
ok good catch, addressed in 9015bff